cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsPipelineSpec.java

```diff
@@ -40,6 +40,10 @@ public class DataStreamsPipelineSpec extends PipelineSpec {
   @Deprecated
   private final String checkpointDirectory;
   private final String pipelineId;
+  private final long maxRetryTimeInMins;
```

Comment:
lets put these all into some sort of retry object. Nicer to group them together, especially if we want to add more configs for it in the future

Reply:
looks like this was done in other places with StreamingRetrySettings, think it makes sense to use that here too.

Reply:
Done.